src/test/kotlin/org/rust/ide/annotator/RsErrorAnnotatorTest.kt

```diff
@@ -4149,4 +4149,44 @@ class RsErrorAnnotatorTest : RsAnnotatorTestBase(RsErrorAnnotator::class) {
         use foo::T1 as _;
         use bar::T2 as _;
     """)
+
+    fun `test no E0537 valid cfg`() = checkErrors("""
+        #[cfg(any(foo, bar))]
+        #[cfg(all(foo, baz))]
+        #[cfg(not(foo))]
+        #[cfg(all(not(foo)))]
+        #[cfg(not(any(foo)))]
+        #[cfg(foo)]
+        fn foo() {}
+    """)
+
+    fun `test E0537 invalid cfg`() = checkErrors("""
+        #[cfg(<error descr="Invalid predicate `an` [E0537]">an(foo)</error>)]
+        #[cfg(<error descr="Invalid predicate `allx` [E0537]">allx(foo)</error>)]
+        #[cfg(<error descr="Invalid predicate `non` [E0537]">non(foo)</error>)]
+        #[cfg(<error descr="Invalid predicate `non` [E0537]">non(an(foo))</error>)]
+        #[cfg(not(<error descr="Invalid predicate `foo` [E0537]">foo()</error>))]
+        #[cfg(all(not(any(bar, <error descr="Invalid predicate `baz` [E0537]">baz()</error>))))]
```

Comment:
I noticed that `rustc` ignores (short-circuits) further predicates, even if they are invalid.
```rust
#[cfg(any(not(y), baz()))]
fn foo() {}
```
This does not produce an error with `rustc`. But I suppose that we want to show an error, since it's invalid and probably a user mistake?

Reply:
Hmm, actually I see that the plugin works similarly.
For this code:
```rust
 #[cfg(all(x, bar()))]
fn foo() {}
```
My annotation marks `bar()`, but the annotation si then not shown (not even in tests), because the whole thing is considered disabled because of short-circuit on `x`.

Reply:
> But I suppose that we want to show an error, since it's invalid and probably a user mistake?

Yep, most likely users want that their code works with any cfg options/features

Reply:
I wonder how to do that though. Because when the rest of the cfg is disabled, it is automatically greyed out and annotations are missing. Some special case in the annotator to ignore cfg-disabling in this case?

Reply:
We can add `checkCfg` flag to `org.rust.lang.utils.RsDiagnosticKt#addToHolder(org.rust.lang.utils.RsDiagnostic, org.rust.ide.annotator.RsAnnotationHolder)` method with `true` as default value
@vlad20012 WDYT?

Reply:
Yep, sounds good to me.

Reply:
Done.